src/main/kotlin/skytils/skytilsmod/features/impl/misc/LockOrb.kt

```diff
@@ -37,30 +39,38 @@ class LockOrb {
         val heldOrb = PowerOrbs.getPowerOrbMatchingItemId(itemId) ?: return
         val orbs = mc.theWorld.getEntities(
             EntityArmorStand::class.java
-        ) { entity: EntityArmorStand? ->
-            val orb = PowerOrbs.getPowerOrbMatchingName(entity!!.displayName.formattedText)
-            orb != null
+        ) {
+            it?.let { PowerOrbs.getPowerOrbMatchingName(it.displayName.formattedText) } != null
         }
         for (orbEntity in orbs) {
-            if (orbEntity == null) return
-            val orb = PowerOrbs.getPowerOrbMatchingName(orbEntity.displayName.formattedText)
-            if (orb != null && orb.ordinal >= heldOrb.ordinal) {
-                if (orbEntity.getDistanceSqToEntity(mc.thePlayer) <= orb.radius.pow(2.0)) {
-                    mc.thePlayer.playSound("random.orb", 0.8f, 1f)
-                    event.isCanceled = true
-                    break
-                }
-            }
+            orbEntity ?: continue
+            val orb = PowerOrbs.getPowerOrbMatchingName(orbEntity.displayName.formattedText) ?: continue
+            if (orb.ordinal < heldOrb.ordinal) continue
+            if (!orb.isPlayerInRange(orbEntity, mc.thePlayer)) continue
+            val remainingDuration = orb.getDurationRemaining(orbEntity)
```

Comment:
?: continue maybe, merge the above 2 if's idk

Reply:
Would be a change in behaviour. rn if it can't find a duration it just ignores the duration settings and blocks placing a new orb anyway. can be changed if you want it that way, should only matter if hypickel changes their format